l10n_es_aeat_mod190/models/mod190.py

```diff
@@ -0,0 +1,566 @@
+
+from odoo import api, fields, models, exceptions, _
+
+
+class L10nEsAeatMod190Report(models.Model):
+
+    _description = 'AEAT 190 report'
+    _inherit = 'l10n.es.aeat.report.tax.mapping'
+    _name = 'l10n.es.aeat.mod190.report'
+    _aeat_number = '190'
+    _period_quarterly = False
+    _period_monthly = False
+    _period_yearly = True
+
+    casilla_01 = fields.Integer(string="[01] Recipients", readonly=True)
+    casilla_02 = fields.Float(string="[02] Amount of perceptions")
+    casilla_03 = fields.Float(string="[03] Amount of retentions")
+    partner_record_ids = fields.One2many(
+        comodel_name='l10n.es.aeat.mod190.report.line',
+        inverse_name='report_id', string='Partner records', ondelete='cascade')
+    registro_manual = fields.Boolean(string='Manual records', default=False)
+    calculado = fields.Boolean(string='Calculated', default=False)
+
+    @api.multi
+    def _check_report_lines(self):
+        """Checks if all the fields of all the report lines
+        (partner records) are filled """
+        for item in self:
+            for partner_record in item.partner_record_ids:
+                if not partner_record.partner_record_ok:
```

Comment:
Cuando hay dos for mejor utilizar mapped, y si hay un if dentro de un for, mejor un filtered o plantearse hacer un search

Reply:
No estoy de acuerdo, de esta forma con el primer error ya falla. En caso contrario debemos computar todo el filtered (que es un for) y luego comprobamos con el mapped (otro for) y hasta que no acaban los dos fors no lanza el error